sw/host/opentitantool/src/command/lc.rs

```diff
@@ -8,19 +8,56 @@ use hex::decode;
 use humantime::parse_duration;
 use serde_annotate::Annotate;
 use std::any::Any;
+use std::rc::Rc;
 use std::time::Duration;
 
 use opentitanlib::app::command::CommandDispatch;
 use opentitanlib::app::TransportWrapper;
 use opentitanlib::dif::lc_ctrl::{DifLcCtrlState, LcCtrlReg};
-use opentitanlib::io::jtag::{JtagParams, JtagTap};
+use opentitanlib::io::jtag::{Jtag, JtagParams, JtagTap};
 use opentitanlib::test_utils::lc_transition::{trigger_lc_transition, trigger_volatile_raw_unlock};
 
 #[derive(serde::Serialize)]
 pub struct LcStateReadResult {
     pub lc_state: DifLcCtrlState,
 }
 
+/// Read, decode, and check the LC state is Raw.
+fn check_lc_state_is_raw(jtag: Rc<dyn Jtag>) -> Result<()> {
+    let lc_state = DifLcCtrlState(jtag.read_lc_ctrl_reg(&LcCtrlReg::LcState)?);
+    if lc_state != DifLcCtrlState::Raw {
+        return Err(anyhow!(
+            "Device must be in Raw LC state. It is in: {}",
+            lc_state
+        ));
+    }
+    Ok(())
+}
+
+/// Parses an unlock token string.
+fn parse_unlock_token_str(token: &str) -> Result<[u32; 4]> {
+    let hex_str_no_sep = token.replace('_', "");
+    let hex_str_prefix = "0x";
+    let sanitized_hex_str = if token.starts_with(hex_str_prefix) {
+        hex_str_no_sep.strip_prefix(hex_str_prefix).unwrap()
+    } else {
+        hex_str_no_sep.as_str()
+    };
+    let token_bytes_vec = decode(sanitized_hex_str)?;
+    let token_word_vec: Vec<u32> = token_bytes_vec
+        .chunks_exact(std::mem::size_of::<u32>())
+        .map(|chunk| u32::from_be_bytes(chunk.try_into().unwrap()))
+        .rev()
+        .collect();
+    if token_word_vec.len() < 4 {
```

Comment:
```suggestion
    if token_word_vec.len() != 4 {
```

Reply:
See @nbdd0121's suggestion below - this one wouldn't catch tokens of length [17,20]

Reply:
fixed.